### ISSUE TITLE
💅 `useExpect` false positive with `expect-type`'s type testing `expectTypeOf`

### ISSUE THREAD

**Issue by Bertie690** [biomejs/biome#9168]
### Environment information

<details>

```bash

N/A

```

</details>

### Rule name

lint/nursery/useExpect

### Playground link

https://biomejs.dev/playground/?lintRules=useExpect&code=aQBtAHAAbwByAHQAIAB7ACAAZABlAHMAYwByAGkAYgBlACwAIABlAHgAcABlAGMAdABUAHkAcABlAE8AZgAsACAAaQB0ACAAfQAgAGYAcgBvAG0AIAAiAHYAaQB0AGUAcwB0ACIAOwAKAAoAaQB0ACgAIgB0AGUAcwB0ACIALAAgACgAKQAgAD0APgAgAHsACgAgACAAZQB4AHAAZQBjAHQAVAB5AHAAZQBPAGYAPAAxAD4AKAApAC4AbgBvAHQALgB0AG8ARQBxAHUAYQBsAFQAeQBwAGUATwBmACgAMgApADsACgB9ACkAOwAKAA%3D%3D

### Expected result

It should not error, as `expectTypeOf` (which vitest re-exports from [`expect-type`](https://github.com/mmkal/expect-type)) is _an_ assertion (albeit not a runtime one).

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
It should be easy to support that. PRs welcome
